packages/callback-handler-svc/src/server/index.ts

```diff
@@ -0,0 +1,155 @@
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Kevin Leyow <kevin.leyow@infitx.com>
+
+ --------------
+ ******/
+
+// workaround for lack of typescript types for mojaloop dependencies
+// eslint-disable-next-line @typescript-eslint/triple-slash-reference
+/// <reference path="../../ambient.d.ts"/>
+/* eslint-disable camelcase */
+import express from 'express'
+import http from 'http'
+import Config from '../shared/config'
+import Logger from '@mojaloop/central-services-logger'
+import Metrics from '@mojaloop/central-services-metrics'
+import { logger } from '~/shared/logger'
+
+type TracestateMap = {
+  tx_end2end_start_ts: number | undefined;
+  tx_callback_start_ts: number | undefined;
+}
+const app = express()
+let appInstance: http.Server
+
+function getTraceStateMap (headers: any): TracestateMap {
+  const tracestate: string = headers.tracestate
+  if (tracestate === undefined) {
+    return {
+      tx_end2end_start_ts: undefined,
+      tx_callback_start_ts: undefined
+    }
+  }
+  let tracestates = {}
+  tracestate
+    .split(',')
+    .map(ts => ts.split('='))
+    .map(([k, v]) => {
+      return { [k]: Number(v) }
+    })
+    .forEach(ts => {
+      tracestates = { ...tracestates, ...ts }
+    })
+  return tracestates as TracestateMap
+}
+
+async function run (): Promise<void> {
+  logger.info(Config)
+  if (!Config.INSTRUMENTATION.METRICS.DISABLED) {
+    Metrics.setup(Config.INSTRUMENTATION.METRICS.config)
+  }
+
+  app.use(express.json())
+  app.get('/health', (_req, res) => {
+    res.json({
+      status: 'OK'
+    })
+  })
+  app.get('/metrics', async (_req, res) => {
+    res.status(200)
+    res.send(await Metrics.getMetricsForPrometheus())
+  })
+  app.get(['/:operation', '/:operation/*'], async (req, res) => {
```

Comment:
What about `PUT` methods?

This is a "Callback Handler" after-all.

Should we not be using `app.all`(ref: https://expressjs.com/en/4x/api.html#app.all) instead of `app.get`?

Alternatively, we could move the logic from the handler here to a re-usable function, and just define multiple `app.<METHODS>`, calling the same implementation.

Let me know your thoughts.

Reply:
Mistake on my part. Switched to `all`.